tests/testbed_validation_rules.py

```diff
@@ -0,0 +1,80 @@
+from abc import ABC, abstractmethod
+import re
+from typing import Sequence, Tuple
+from pyats.topology.testbed import Testbed
+
+
+class Rule(ABC):
+
+    ip_pattern = r"^\d{3}\.\d{3}\.\d{1,3}\.\d{1,3}$"
+
+    @abstractmethod
+    def is_passed(self, testbed: Testbed) -> bool:
+        """Provides a result of rule execution"""
+        pass
+
+    @abstractmethod
+    def __str__(self) -> str:
+        """Returns rule name"""
+        pass
+
+
+class CheckUserCredentials(Rule):
+    def is_passed(self, testbed: Testbed) -> bool:
+        for device in testbed:
+            device.connect()
+            assert device.connected
+        return True
+
+    def __str__(self) -> str:
+        return "Check user credentials"
+
+
+class CheckDevicesIp(Rule):
+    def is_passed(self, testbed: Testbed) -> bool:
+        ips = [str(device.connections.main.ip) for device in testbed]
+        ips_validations = list(map(lambda ip: bool(re.fullmatch(self.ip_pattern, ip)), ips))
+        return all(ips_validations)
+
+    def __str__(self) -> str:
+        return "Check devices ip format"
+
+
+class CheckSeleniumGridUrl(Rule):
+
+    url_pattern = r"^https?:\/\/.+:\d+$"
+
+    def is_passed(self, testbed: Testbed) -> bool:
+        grid_url = testbed.custom["selenium-grid"]
+        return bool(re.fullmatch(self.url_pattern, grid_url))
+
+    def __str__(self) -> str:
+        return "Check Selenium grid url format"
+
+
+class CheckServerIp(Rule):
+    def is_passed(self, testbed: Testbed) -> bool:
+        server_ip = testbed.servers.server_alias.address
+        assert re.fullmatch(self.ip_pattern, server_ip)
```

Comment:
@SergiyBeno  Is it better?
```suggestion
        return re.fullmatch(self.ip_pattern, server_ip)
```

Reply:
Fixed